news/5395.bugfix.rst

```diff
@@ -0,0 +1 @@
+Change validation schema for Requires object to allow for both ``python_version`` and ``python_full_version``. Also added regex validation so that it must be a valid python version and python_version is like 3.10, and python_full_version is like 30.10.7
```

Comment:
What about version like `3.11.0rc3` or `dev` or `my-local-patch`.

I actually dont think this is necessary to have the regex.

But I do think the schema should validate the python or python_full_version exist like I commented in the issue in pipenv.

Reply:
@oz123 really good point and good call out. I updated and ended up just removing the "excludes" from the validation. If i understand correctly I think the point you were trying to make with cerebus here was to make sure at least one of the python version fields is added when the object is initialized.

unfortunately I played around quite a bit and couldn't figure out within the current way it is being handled.
The `oneof`, or `anyof` would plug in well if the field was a dict, but the type is string and so couldn't figure out how to get it to validate that if it doesn't exist that at least the other version type was incluided when the `Requires` object is initialized.

So now: it will accept both `python_version` and `python_full_version` but it will also initialize is an empty object is passed to `Requires`.

If we want to enforce that there is at least one type of python version in the initialize field we probably need to plug that in the validate() method I would guess. 

Unless I missed something completely.

Reply:
Actually, the fields `python_version` and `python_version` are optional in a `Pipefile` so it's ok that it validates. If you specify a python version you convey that you tested the project with a certain version, and that is fine.